### ISSUE TITLE
[3.3.5] Core/Movement MOVEMENTFLAG_CAN_FLY not removed correctly in some cases

### ISSUE THREAD

**Issue by sirikfoll** [TrinityCore/TrinityCore#16670]
**Description**: MOVEMENTFLAG_CAN_FLY is not removed from players if they enter instances while on a flying mount, this results in players being allowed to fly without mounts.

**Current behaviour**: unmounted players can fly

**Expected behaviour**: unmounted players shouldn't fly

**Steps to reproduce the problem**:

1. Set worldserver.conf -> Compression = 9 or run in debug mode
2. .tele blacktemple
3. .cast 72807
4. Fly through the portal
5. press "espace key" and fly without mounts
6. If not worked, try again, it's not a 100% chance

**Branch(es)**: 335(not tested in 6x)

**TC hash/commit**: dab661c8dad6

**TDB version**: 335.60

**Operating system**: Windows 10

[//]: # (This template is for problem reports, for other type of reports edit it accordingly)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)

**Comment by Nawuko** [comment]
I can reproduce this (most of the time 2-3 out of 10 tries). This dosent have todo with Packet compression. The Fly aura is removed from the narrow entrance not from the zone change. The Server has correctly removed the flag (so its not an Spell-System or Unit object problem) but the client rather ignores the un-fly package during map change.
Let me summon @Shauren cause he is the one-and-only when it comes to package stuff :+1: (jk)

TL:DR
Can confirm, Latency Problem, Client dosent get unfly package.

--
sorry for my english ....

**Comment by enkrii** [comment]
This is not true for regular player accounts. Using a GM account simply skips this cleanup for convenience (`.gm fly on`).

```c++
    REMOVE_VIOLATING_FLAGS(mi->HasMovementFlag(MOVEMENTFLAG_FLYING | MOVEMENTFLAG_CAN_FLY) && GetSecurity() == SEC_PLAYER &&
        !mover->HasAuraType(SPELL_AURA_FLY) &&
        !mover->HasAuraType(SPELL_AURA_MOD_INCREASE_MOUNTED_FLIGHT_SPEED),
        MOVEMENTFLAG_FLYING | MOVEMENTFLAG_CAN_FLY);
```
